packages/api/src/command/medical/admin/hie-overview.ts

```diff
@@ -0,0 +1,240 @@
+  patient: HiePatient;
+};
+
+type CqPdErrorConsolidated = {
+  error: string;
+  gatewayNames: string[];
+};
+
+type HiePatientOverview = {
+  cxId: string;
+  patientId: string;
+  patientData: PatientData;
+  carequality: {
+    links: CQLink[];
+    success: CqPdSuccessResponse[];
+    errors: CqPdErrorConsolidated[];
+  };
+};
+
+type DbQueryResponseRecord = {
+  cx_id: string;
+  id: string;
+  patient_data: PatientData;
+  cq_links: { links: CQLink[] };
+  response_data: OutboundPatientDiscoveryResp | undefined;
+  name: string | undefined | null;
+};
+
+export async function getHieOverview(
+  patientId: string,
+  debugLevel: DebugLevel
+): Promise<HiePatientOverview | undefined> {
+  const queryResp: DbQueryResponseRecord[] = await getDataFromDb(patientId, debugLevel);
+  const [respErrors, respSuccess] = partition(
+    queryResp,
+    r => r.response_data?.patientMatch === false
+  );
+  const patientData = respSuccess[0]?.patient_data;
+
+  const errors: CqPdErrorConsolidated[] = respErrors.reduce(
+    (acc: CqPdErrorConsolidated[], row: DbQueryResponseRecord) => {
+      const rowErrorDetail = (
+        row.response_data?.operationOutcome?.issue[0]?.details as { text: string | undefined }
+      )?.text;
+      if (!rowErrorDetail) return acc;
+      const gatewayName = getGatewayName(row);
+      const existingErrorEntry = acc.find(a => a.error === rowErrorDetail);
+      if (existingErrorEntry) {
+        existingErrorEntry.gatewayNames.push(gatewayName);
+        return acc;
+      }
+      acc.push({
+        error: rowErrorDetail,
+        gatewayNames: [gatewayName],
+      });
+      return acc;
+    },
+    new Array<CqPdErrorConsolidated>()
+  );
+
+  const hieOverview = respSuccess.reduce(
+    (acc: HiePatientOverview | undefined, row: DbQueryResponseRecord) => {
+      if (!row.response_data) return acc;
+      const patientOfSuccessfulResp = responseToPatient(row.response_data, patientData);
+      if (!patientOfSuccessfulResp) {
+        console.log(`Skipping response with missing patient data: ${row.response_data.id}`);
+        return acc;
+      }
+      const gatewayName = getGatewayName(row);
+      const gateway = responseToGateway(row.response_data, gatewayName);
+      const success = {
+        patient: patientOfSuccessfulResp,
+        gateway,
+      };
+      if (acc) {
+        acc.carequality.success.push(success);
+        return acc;
+      }
+      const cqLinks = responseToCqLinks(row);
+      return {
+        cxId: row.cx_id,
+        patientId: row.id,
+        patientData,
+        carequality: {
+          links: cqLinks,
+          success: [success],
+          errors,
+        },
+      };
+    },
+    undefined
+  );
+  return hieOverview;
+}
+
+async function getDataFromDb(
+  patientId: string,
+  debugLevel: DebugLevel
+): Promise<DbQueryResponseRecord[]> {
+  const matchQuery = debugLevel === "success" ? `and r.data->>'patientMatch' = 'true'` : "";
+  const query =
```

Comment:
nit: I find sql easier to follow if we could have more descriptive aliases instead of p or d but not a big deal

Reply:
And maybe an example here as comment of the expected output, cause if not i have to draw up a mental model of the output to reference throughout the rest of the code.

Reply:
agree. I would prefer `pdr` for patient_discovery_result over `r`

Reply:
Aliases are supposed to be small, otherwise we might as well write the table name.  I added more letters to some aliases using @RamilGaripov's suggestions, I think it's easier now. Let me know.